alpenhorn/client.py

```diff
@@ -853,34 +854,64 @@ def import_files(node_name, verbose, acq, dry):
         print("Unknown node.")
         return
 
-    # Construct list of directories that might be acquisitions
+    cwd = os.getcwd()
+    # Construct a dictionary of directories that might be acquisitions and the of
+    # list files that they contain
+    db_acqs = ac.ArchiveAcq.select(ac.ArchiveAcq.name)
+    acq_files = defaultdict(list)
     if len(acq) == 0:
-        db_acqs = dict(ac.ArchiveAcq
-                       .select(ac.ArchiveAcq.name, ac.ArchiveAcq.id)
-                       .tuples())
-
-        acqs = []           # Directories which are known acquisitions
-        for d, ds, fs in os.walk(os.getcwd()):
-            d = os.path.relpath(d)
-            if d == '.':            # skip the current directory
+        tops = [cwd]
+    else:
+        db_acqs = db_acqs.where(ac.ArchiveAcq.name >> acq)
+        tops = []
+        for acq_name in acq:
+            acq_dir = os.path.join(node.root, acq_name)
+            if not os.path.isdir(acq_dir):
+                print('Aquisition "%s" does not exist in this node. Ignoring.' % acq_name)
+                continue
+            if acq_dir == cwd:
+                # the current directory is one of the limiting acquisitions, so
+                # we can ignore all others in the `--acq` list
+                tops = [acq_dir]
+                break
+            elif cwd.startswith(acq_dir):
+                # the current directory is inside one of the limiting
+                # acquisitions, so we can just walk its subtree
+                tops = [cwd]
+                break
+            elif acq_dir.startswith(cwd):
+                # the acquisition is inside the current directory, so we can
+                # just walk its subtree
+                tops.append(acq_dir)
+            else:
+                # TODO: confirm this is what we want to do in this case!
```

Comment:
This seems like the right choice to me here, so let's go with it for now and see how it works out.